packages/blade/src/utils/styled-props/getStyledProps.tsx

```diff
@@ -0,0 +1,72 @@
+import type { StyledProps } from '../../components/Box/BaseBox/types';
+
+/**
+ * This utility takes all your props and returns only the styled props that are to be used on components
+ *
+ * ### Usage
+ *
+ * ```tsx
+ * import type { StyledProps } from '~utils';
+ * import { getStyledProps } from '~utils';
+ *
+ * const MyComponent = (props: MyComponentProps & StyledProps) => {
+ *  return (
+ *    <BaseBox {...getStyledProps(props)}>
+ *      // Your component code
+ *    </BaseBox>
+ *  )
+ * }
+ * ```
+ *
+ *
+ * Or if you don't want to wrap component in BaseBox, checkout [useStyledProps](./useStyledProps.web.tsx)
+ *
+ * ```tsx
+ * import { useStyledProps } from '~utils';
+ *
+ * const TextComponentWithStyledProps = styled.p((props) => {
+ *  const styledPropsCSSObject = useStyledProps(props)
+ *
+ *  return {
+ *    ...styledPropsCSSObject,
+ *    ...your other CSS,
+ *  }
+ * })
+ * ```
+ *
+ * Checkout implementation in `BaseText` components for more details
+ *
+ * - Web: [`BaseText.web.tsx`](../../components/Typography/BaseText/BaseText.web.tsx)
+ * - Native: [`BaseText.native.tsx`](../../components/Typography/BaseText/BaseText.native.tsx)
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getStyledProps = (props: Record<string, any>): StyledProps => {
```

Comment:
What's the difference between `getStyledProps` & `useStyledProps`. How will we know when to use what? 🤔

Reply:
Have added it in JSDoc of this function. We use `getStyledProps` when we are adding Box wrapper, and `useStyledProps` when we're adding props to component itself

Reply:
so `useStyledProps` is just a memoised version for `getStyledProps` if i get it correctly based on what you explained last time?

Reply:
Yup